src/main/java/uk/gov/hmcts/reform/civil/validation/UnavailableDateValidator.java

```diff
@@ -50,39 +51,32 @@ public boolean isValid(UnavailableDate value, ConstraintValidatorContext context
         return errors;
     }
 
+    public List<String> validateFastClaimHearing(HearingLRspec hearingLRspec) {
+        List<String> errors = new ArrayList<>();
+        if (hearingLRspec.getUnavailableDatesRequired() == YES && isFastClaimHearingNullOrEmpty(hearingLRspec)) {
```

Comment:
I checked the method isFastClaimHearingNullOrEmpty and its not checking null or empty for hearing object but just checking the null of unavailabledates, so reversing the check here would not cover against the nullpointer coming on hearingLRSpec object.

I think this condition and further similar conditions should be done something like this

`if (hearingLRspec != null && hearingLRspec.getUnavailableDatesRequired() == YES && isFastClaimHearingNullOrEmpty(hearingLRspec)) {`

Reply:
change incorporated